### PR TITLE
Add XML parser for generic webhook payloads

### PR DESCRIPTION
**Pull request by Half-Shot**
Does what it says on the tin. We also now document what formats we support in the documentation.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
     "uuid": "^8.3.2",
     "vm2": "^3.9.6",
     "winston": "^3.3.3",
+    "xml2js": "^0.4.23",
```
Comment (Half-Shot) [inline package.json:65]: I chose this because we already use it in the rss parser.

File: src/generic/Router.ts [hunk 2]
```diff
@@ -55,13 +55,34 @@ export class GenericWebhooksRouter {
...
+        });
+    }
+
     public getRouter() {
         const router = Router();
         router.all(
             '/:hookId',
-            express.text({ type: 'text/*'}),
+            GenericWebhooksRouter.xmlHandler,
             express.urlencoded({ extended: false }),
```
Comment (AndrewFerr) [inline src/generic/Router.ts:83]: Is this for `x-www-form-urlencoded`?  And whatever it is, is it a distinct enough type to be documented in the "Payload formats" section?

**Review by AndrewFerr — CHANGES_REQUESTED** [review]

File: src/generic/Router.ts (hunk 2 above)
Comment (Half-Shot) [reply src/generic/Router.ts:83]: Good point. I think I missed it out.

File: src/provisioning/provisioner.ts [hunk 3]
```diff
@@ -70,7 +70,6 @@ export class Provisioner {
             (...args) => this.checkUserPermission("write", ...args),
             this.deleteConnection.bind(this),
         );
-        this.expressRouter.use((err: unknown, req: Request, res: Response, next: NextFunction) => errorMiddleware(log)(err, req, res, next));
```
[the remark this answers is not in the document]
Comment (Half-Shot) [reply src/provisioning/provisioner.ts:73]: Failed to do a Ctrl-S, oops.

File: src/generic/Router.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Half-Shot) [reply src/generic/Router.ts:83]: Ah yes, most probably!

**Review by AndrewFerr — APPROVED** [review]

**Pull request merged**